compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -99,6 +99,13 @@ pub(crate) struct AcirDynamicArray {
     block_id: BlockId,
     /// Length of the array
     len: usize,
+    /// An ACIR dynamic array is a flat structure, so we use
```

Comment:
Isn't the type of all elements in an array the same?

I might be reading this wrong

Reply:
This is true for a homogenous array but not necessarily true for a nested array. If you look at `array_set_value` dynamic arrays only consider a witness index to write into and a witness value which is being written. We lose the type information of each `AcirValue::Var` once we start operating over just an `AcirDynamicArray`. This has been fine up until now as reading/writing fields has served our needs. However, now that we let you pass a dynamic array as input into a black box function the numeric type information matters as we may possibly be writing the incorrect number of bits. Most foreign hash function inputs are specified by just a witness index and a number of bits.

Reply:
It would probably be useful to add a regression test where I pass a dynamic array to a blackbox that is fetched from inside a nested dynamic array. I'll do that to display why we can't just have a single numeric type here.